server/models/user.rb

```diff
@@ -33,4 +33,9 @@ def password_matches?(raw_password)
   def self.public_attrs
     [:username, :id]
   end
+  def avatar_url
+    hash = Digest::MD5.hexdigest(if self.email then self.email.strip.downcase else "" end)
+    self.avatar ? "/api/auth/users/#{self.id}/avatar"
```

Comment:
The motivation for placing this logic in the model is to centralize the logic for swapping between gravatar and the user's uploaded avatar. We could move it to a helper somewhere, but I think that will actually just obfuscate the underlying behavior. I googled around for a `link_to`, but there doesn't appear to be a native one for sintra.